app/models/agents/http_status_agent.rb

```diff
@@ -46,29 +50,47 @@ def validate_options
       errors.add(:base, "a url must be specified") unless options['url'].present?
     end
 
+    def header_array(str)
+      (str || '').split(',').map(&:strip)
+    end
+
     def check
-      check_this_url interpolated[:url]
+      check_this_url interpolated[:url], header_array(interpolated[:headers_to_save])
     end
 
     def receive(incoming_events)
       incoming_events.each do |event|
         interpolate_with(event) do
-          check_this_url interpolated[:url]
+          check_this_url interpolated[:url], header_array(interpolated[:headers_to_save])
         end
       end
     end
 
     private
 
-    def check_this_url(url)
+    def check_this_url(url, local_headers)
+      # Track time
       measured_result = TimeTracker.track { ping(url) }
+
+      payload = { 'url' => url, 'response_received' => false, 'elapsed_time' => measured_result.elapsed_time }
+
+      # Deal with failures
       if measured_result.result
-        create_event payload: { 'url' => url, 'status' => measured_result.status.to_s, 'response_received' => true, 'elapsed_time' => measured_result.elapsed_time }
+        payload.merge!({ 'response_received' => true, 'status' => measured_result.status.to_s })
+        # Deal with headers
+        if local_headers.present?
+          header_results = measured_result.result.headers.select {|header, value| local_headers.include?(header)}
```

Comment:
This way, header names are compared case-sensitively, so you have to specify header names in `headers_to_save` to match exactly with what's returned by Faraday.  The specs are wrong (but passing due to mocking) in that they are using `SecureRandom.uuid` as header name which consists of lower case letters and digits, where an actual response will have capitalized header names, which won't match each other.

This should be `measured_result.result.headers.slice(*local_headers)` instead and the specs need to be fixed.